plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/RequestSampler.java

```diff
@@ -18,21 +24,44 @@
 import io.opentelemetry.sdk.trace.samplers.Sampler;
 import io.opentelemetry.sdk.trace.samplers.SamplingResult;
 
+import static org.opensearch.telemetry.tracing.AttributeNames.TRACE;
+
 /**
- * HeadBased sampler
+ * RequestSampler based on HeadBased sampler
  */
 public class RequestSampler implements Sampler {
-    private final Sampler defaultSampler;
+    private final HashMap<String, Sampler> samplers;
+    private final TelemetrySettings telemetrySettings;
+    /**
+     * Logger instance for logging messages related to the request sampler.
+     */
+    protected Logger logger;
 
-    // TODO: Pick value of TRACE from PR #9415.
-    private static final String TRACE = "trace";
+    /**
+     * Creates request sampler which applies based on all applicable sampler
+     * @param telemetrySettings TelemetrySettings
+     */
+    public RequestSampler(TelemetrySettings telemetrySettings) {
+        this.telemetrySettings = telemetrySettings;
+        this.samplers = new HashMap<>();
+        this.logger = LogManager.getLogger(getClass());
+        this.SamplerInit();
+    }
 
     /**
-     * Creates Head based sampler
-     * @param defaultSampler defaultSampler
+     * Initialises all samplers based on telemetry setting
      */
-    public RequestSampler(Sampler defaultSampler) {
-        this.defaultSampler = defaultSampler;
+    private void SamplerInit() {
+        for (String samplerName : this.telemetrySettings.getSamplingOrder()) {
```

Comment:
Please extract the logic of instantiating sampler chain into `OTelSamplerFactory` (see please `OTelSpanExporterFactory` as an example of similar implementation for `SpanExporter`s)

Reply:
Just to confirm, we just to move creation from `SamplerInit()` to OTelSpanExporterFactory class. If yes, then can you help me understand how is it helpful here ? Here also the constructor of RequestSampler will only execute once. Is it from point of redability ?